### PR TITLE
[CK_TILE] support group from cmdline

### PR DESCRIPTION
**Pull request by carlushuang**

### CODE REVIEW

File: example/ck_tile/01_fmha/utils.hpp [hunk 1]
```diff
@@ -82,6 +83,74 @@ std::vector<int32_t> generate_seqstarts(mode_enum mode,
...
+ *   q_val=1,2   k_val=4     -> not OK, k must have same splits with q
+ */
+std::tuple<std::vector<ck_tile::index_t>, std::vector<ck_tile::index_t>>
+decode_seqlen(mode_enum mode,
+              ck_tile::index_t batch,
+              std::string q_val,
+              std::string k_val,
+              std::optional<unsigned> seed = std::nullopt)
+{
+#define _S2I_(str_) static_cast<ck_tile::index_t>(atoi((str_).c_str()))
```
Comment (poyenc) [inline example/ck_tile/01_fmha/utils.hpp]: I think we can use `std::stoi()` here

**Comment by poyenc** [comment]
except the function usage, other code LGTM

File: example/ck_tile/01_fmha/fmha_fwd.cpp [hunk 2]
```diff
@@ -177,10 +179,16 @@ bool run(const ck_tile::ArgParser& arg_parser)
...
     }
 
-    ck_tile::index_t seqlen_q = arg_parser.get_int("s");
-    ck_tile::index_t seqlen_k = arg_parser.get_int("s_k");
-    if(seqlen_k < 0)
-        seqlen_k = seqlen_q;
+    auto [seqlen_qs, seqlen_ks] =
+        decode_seqlen(mode, batch, arg_parser.get_str("s"), arg_parser.get_str("s_k"));
+
+#if 0
```
Comment (poyenc) [inline example/ck_tile/01_fmha/fmha_fwd.cpp:194]: do we need a macro to turn on/off debug prints?

File: include/ck_tile/core/arch/amd_buffer_addressing.hpp [hunk 3]
```diff
@@ -29,6 +29,25 @@ CK_TILE_DEVICE int32x4_t make_wave_buffer_resource(const void* ptr, uint32_t siz
     return __builtin_bit_cast(int32x4_t, res);
 }
 
+namespace impl {
+// below type indicate the data type used for buffer load inline asm
+// clang-format off
+template<index_t N, typename T> struct buffer_load_trait;
+
+template<typename T> struct buffer_load_trait<16, T> { using type = fp32x4_t; };
```
Comment (poyenc) [inline include/ck_tile/core/arch/amd_buffer_addressing.hpp]: can we name the member type `payload_t` or something similar?

File: example/ck_tile/01_fmha/fmha_fwd.cpp (hunk 2 above)
Comment (carlushuang) [reply example/ck_tile/01_fmha/fmha_fwd.cpp:194]: that is for debug purpose :)

**Review by poyenc — DISMISSED** [review]

**Review by poyenc — DISMISSED** [review]

**Review by zjing14 — APPROVED** [review]

**Review by poyenc — APPROVED** [review]

**Pull request merged**